### ISSUE TITLE
"Let each entry choose which sites it should be saved to" doesn't seem to work with a multi-site setup Feed.

### ISSUE THREAD

**Issue by jan-dh** [craftcms/feed-me#1279]
### Description

I have a question about Feed Me which puzzles me a bit. I have a blog channel set up for multiple sites with the option: Let each entry choose which sites it should be saved to . I set up 2 feeds, one for Site A, the other for site B and I feed them some blogposts.  They both have an old_id field. If a blogpost has the same old_id value for the 2 different sites I would expect them to be linked together but that's not the case. 🙃 Is this a bug? (Feedme not taking into account the new custom option)

**Comment by brandonkelly** [comment]
If you know the specific entry IDs that need to be propagated to the new site ahead of time, the following command can be run _before_ running the Feed Me import:

```sh
php craft resave/entries \
  --element-id="1,2,3" \
  --site=oldSiteHandle \
  --propagate-to=newSiteHandle
```

(Note `--propagate-to` was added in Craft 4.4.7.)

But ideally, Feed Me should be aware when sections are using the `custom` propagation method, and automatically propagate any entries that are missing from the import’s target site.

Missing entries can be included in the import’s entry query like so:

```php
if ($section->propagationMethod === Section::PROPAGATION_METHOD_CUSTOM) {
    $entryQuery
        ->site('*')
        ->preferSites([$targetSiteId])
        ->unique();
}
```

Then, when iterating through the entries, entries can be propagated when necessary like so:

```php
// Did the entry come back in a different site?
if ($entry->siteId != $targetSiteId) {
    // Skip it if its section doesn't use the `custom` propagation method
    if ($entry->getSection->propagationMethod !== Section::PROPAGATION_METHOD_CUSTOM) {
        continue;
    }

    // Give the entry a status for the import's target site
    // (This is how the `custom` propagation method knows which sites the entry should support.)
    $siteStatuses = ElementHelper::siteStatusesForElement($entry);
    $siteStatuses[$targetSiteId] = $entry->getEnabledForSite();
    $entry->setEnabledForSite($siteStatuses);

    // Propagate the entry, and swap $entry with the propagated copy
    $entry = Craft::$app->getEntries()->propagateElement($entry, $targetSiteId);
}
```

(Note that this will require Craft 4.4.8+, as I only [just added](https://github.com/craftcms/cms/commit/76c2d9d8438fe04bfbc3c186af8faa24109a5816) `return $siteElement` to `Elements::propagateElement()`.)

**Comment by angrybrad** [comment]
Fixed in https://github.com/craftcms/feed-me/pull/1292 for the next v5 release.

**Comment by angrybrad** [comment]
We just released Feed Me 5.1.3 with this fix.
